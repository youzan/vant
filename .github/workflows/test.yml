name: CI

on:
  push:
    branches:
      - '**'

  pull_request:
    branches:
      - main

  workflow_dispatch:

jobs:
  lint:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v4

      - name: Install pnpm
        run: |
          npm install -g corepack@latest --force 
          corepack enable

      - uses: actions/setup-node@v4
        with:
          node-version: '22.14.0'
          cache: 'pnpm'

      - name: Install dependencies
        run: pnpm install

      - name: Run linter
        run: npm run lint

  test:
    runs-on: macos-14
    steps:
      - uses: actions/checkout@v4

      - name: Install pnpm
        run: |
          npm install -g corepack@latest --force 
          corepack enable

      - uses: actions/setup-node@v4
        with:
          node-version: '22.14.0'
          cache: 'pnpm'

      - name: Install dependencies
        run: pnpm install

      - name: Run test cases
        run: pnpm run test:coverage

      - name: Upload coverage to Codecov
        uses: codecov/codecov-action@v5
        with:
          token: ${{ secrets.CODECOV_TOKEN }}
          directory: ./packages/vant/test/coverage

  build:
    runs-on: macos-14
    steps:
      - uses: actions/checkout@v4

      - name: Install pnpm
        run: |
          npm install -g corepack@latest --force 
          corepack enable

      - uses: actions/setup-node@v4
        with:
          node-version: '22.14.0'
          cache: 'pnpm'

      - name: Install dependencies
        run: pnpm install

      - name: Build
        run: npm run build
