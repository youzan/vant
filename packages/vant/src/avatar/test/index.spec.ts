import { Avatar } from '..';
import { mount } from '../../../test';

test('should render image avatar correctly', () => {
  const wrapper = mount(Avatar, {
    props: {
      src: 'https://example.com/avatar.jpg',
      alt: 'User Avatar',
    },
  });

  expect(wrapper.html()).toMatchSnapshot();
  const img = wrapper.find('img');
  expect(img.attributes('src')).toBe('https://example.com/avatar.jpg');
  expect(img.attributes('alt')).toBe('User Avatar');
});

test('should render text avatar correctly when no src provided', () => {
  const wrapper = mount(Avatar, {
    props: {
      text: 'AB',
    },
  });

  expect(wrapper.html()).toMatchSnapshot();
  expect(wrapper.text()).toBe('AB');
});

test('should render icon avatar correctly', () => {
  const wrapper = mount(Avatar, {
    props: {
      icon: 'user-o',
    },
  });

  expect(wrapper.html()).toMatchSnapshot();
  const icon = wrapper.find('.van-icon-user-o');
  expect(icon.exists()).toBe(true);
});

test('should render default slot when no props provided', () => {
  const wrapper = mount(Avatar, {
    slots: {
      default: () => 'Custom Content',
    },
  });

  expect(wrapper.html()).toMatchSnapshot();
  expect(wrapper.text()).toBe('Custom Content');
});

test('should apply different sizes correctly', () => {
  const wrapper = mount(Avatar, {
    props: {
      size: 'large',
      text: 'A',
    },
  });

  expect(wrapper.classes()).toContain('van-avatar--large');

  // Test number size
  const wrapper2 = mount(Avatar, {
    props: {
      size: 60,
      text: 'B',
    },
  });

  expect(wrapper2.style.width).toBe('60px');
  expect(wrapper2.style.height).toBe('60px');

  // Test string size
  const wrapper3 = mount(Avatar, {
    props: {
      size: '80px',
      text: 'C',
    },
  });

  expect(wrapper3.style.width).toBe('80px');
  expect(wrapper3.style.height).toBe('80px');
});

test('should apply different shapes correctly', () => {
  const wrapper = mount(Avatar, {
    props: {
      shape: 'square',
      text: 'A',
    },
  });

  expect(wrapper.classes()).toContain('van-avatar--square');
});

test('should apply background color and text color', () => {
  const wrapper = mount(Avatar, {
    props: {
      bgColor: '#ff0000',
      color: '#ffffff',
      text: 'A',
    },
  });

  expect(wrapper.style.backgroundColor).toBe('rgb(255, 0, 0)');
  expect(wrapper.style.color).toBe('rgb(255, 255, 255)');
});

test('should emit error event when image load fails', async () => {
  const wrapper = mount(Avatar, {
    props: {
      src: 'invalid-url.jpg',
    },
  });

  const img = wrapper.find('img');
  await img.trigger('error');

  expect(wrapper.emitted('error')).toHaveLength(1);
});

test('should prioritize src over text and icon', () => {
  const wrapper = mount(Avatar, {
    props: {
      src: 'https://example.com/avatar.jpg',
      text: 'AB',
      icon: 'user-o',
    },
  });

  expect(wrapper.find('img').exists()).toBe(true);
  expect(wrapper.text()).toBe('');
  expect(wrapper.find('.van-icon-user-o').exists()).toBe(false);
});

test('should prioritize text over icon when no src', () => {
  const wrapper = mount(Avatar, {
    props: {
      text: 'AB',
      icon: 'user-o',
    },
  });

  expect(wrapper.text()).toBe('AB');
  expect(wrapper.find('.van-icon-user-o').exists()).toBe(false);
});

test('should handle empty text correctly', () => {
  const wrapper = mount(Avatar, {
    props: {
      text: '',
      icon: 'user-o',
    },
  });

  expect(wrapper.find('.van-icon-user-o').exists()).toBe(true);
});
