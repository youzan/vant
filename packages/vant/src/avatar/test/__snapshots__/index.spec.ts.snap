// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should render default slot when no props provided 1`] = `
<div class="van-avatar van-avatar--round">
  Custom Content
</div>
`;

exports[`should render icon avatar correctly 1`] = `
<div class="van-avatar van-avatar--round">
  <i class="van-avatar__icon van-icon van-icon-user-o">
  </i>
</div>
`;

exports[`should render image avatar correctly 1`] = `
<div class="van-avatar van-avatar--round">
  <img
    src="https://example.com/avatar.jpg"
    alt="User Avatar"
    class="van-avatar__img"
  >
</div>
`;

exports[`should render text avatar correctly when no src provided 1`] = `
<div class="van-avatar van-avatar--round">
  <span class="van-avatar__text">
    AB
  </span>
</div>
`;
